### ISSUE TITLE
display: none causing errors

### ISSUE THREAD

**Issue by acidjazz** [BKWLD/vue-in-viewport-mixin#20]
if i use this mixin in a component, and then put something like `print_hidden` from tailwindcss above or on that component, i get the following error:

```
TypeError: Failed to construct 'IntersectionObserver': The provided double value is non-finite.
    at VueComponent.addInViewportHandlers (index.js:526)
    at VueComponent.inViewportInit (index.js:514)
    at VueComponent.reInitInViewportMixin (index.js:509)
    at VueComponent.inViewportThresholdWithMax (index.js:500)
    at Watcher.run (commons.app.js:14062)
    at flushSchedulerQueue (commons.app.js:13804)
    at Array.<anonymous> (commons.app.js:11485)
    at flushCallbacks (commons.app.js:11411)
```

the line that triggers it:

```js
      this.inViewportObserver = new IntersectionObserver(this.updateInViewport, {
```

**Comment by acidjazz** [comment]
Update: this is only duplicate-able by putting chrome into the `print` media query rendering mode:

![image](https://user-images.githubusercontent.com/967369/61941572-60102180-af5d-11e9-9a94-ae344d44d792.png)

![image](https://user-images.githubusercontent.com/967369/61941590-6a322000-af5d-11e9-9363-f5ec6893f2c3.png)
![image](https://user-images.githubusercontent.com/967369/61941610-70c09780-af5d-11e9-96f6-6b5854cbc206.png)

**Comment by acidjazz** [comment]
I've now been able to reproduce this when chrome is not in any custom rendering mode:

1. I have a component that uses the mixin and has the `watch`

2. I include the component in a parent page/component but i add a class that gives it the `display: none` CSS property.

3. The same error occurs.

I assume  this is something related to chrome not allowing this to be accessible on the DOM since its hidden?

**Comment by clexi** [comment]
i had the same error. If a parent component is set to `display: none` e.g. via `v-show` the errors occurs in the child component which is using the mixin. I changes `v-show` to `v-if` it solves my problem temporary.

**Comment by can-keklik** [comment]
It probably just needs an isFinite check. I'm not very familiar with CoffeeScript, so will try to demonstrate it on js:

https://github.com/BKWLD/vue-in-viewport-mixin/blob/74819bc85eccf1036294656f3c089159fe11c48f/index.js#L198

In some cases, the expression evaluates to NaN.

For instance: `Math.min(1, 0/0)`

So that division should be checked like this:

```js
var heightRatio = root.height / target.height;
if (Number.isFinite(heightRatio)) {
    this.inViewport.maxThreshold = Math.min(1, heightRatio); 
}
else {
    this.inViewport.maxThreshold = 1;
}
```

It worked for me, but you should do your own experiments of course.

**Comment by weotch** [comment]
Thanks for the suggestion @can-keklik,  I released a fix as 2.0.2
